responses/__init__.py

```diff
@@ -511,16 +512,27 @@ def __init__(
         response_callback=None,
         passthru_prefixes=(),
         target="requests.adapters.HTTPAdapter.send",
+        registry=DefaultRegistry,
     ):
         self._calls = CallList()
+        self._registry = registry()
         self.reset()
         self.assert_all_requests_are_fired = assert_all_requests_are_fired
         self.response_callback = response_callback
         self.passthru_prefixes = tuple(passthru_prefixes)
         self.target = target
 
+    def get_registry(self):
+        return self._registry
+
+    def set_registry(self, new_registry):
+        self.reset()
+        self._registry = new_registry()
```

Comment:
I think a change of the Registry class should not reset the registered responses. Instead, the new instance should be initialized with the previously registered responses.
You could easily do that by giving the `BaseRegistry` constructor an optional argument which sets the `_responses` attribute and make it itterable, so you could easily switch between implementations.